src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java

```diff
@@ -52,9 +86,27 @@ public BigInteger safeTimeForTablet(String tabletId) {
         return tabletSafeTime.get(tabletId);
     }
 
+    /**
+     * Retrieves a message from the queue if it is present. The retrieval is based on a condition
+     * that the message in the front of the queue should have a commit time less than or equal to
+     * the stream safetime, an empty {@link Optional} is returned if the condition fails.
+     * @return a {@link Message} wrapped in an {@link Optional}, or an {@link Optional#empty()} if
+     * message is null
+     */
     private Optional<Message> peek() {
         Message message = queue.peek();
-        Optional<Message> peeked = message != null && message.commitTime.compareTo(this.streamSafeTime()) < 0
+
+        if (message == null) {
+            LOGGER.warn("Message after peeking is null (actually means to message in queue)");
+        } else {
+            LOGGER.warn("Message is not null in queue - actual message is {}", message);
+            if (!(message.commitTime.compareTo(this.streamSafeTime()) <= 0)) {
+                LOGGER.warn("Comparison commit time for message and stream safetime failed (may indicate issues)");
```

Comment:
Require a config parameter that after X iterations or time, an assert should be triggered.